hs_core/tests/api/native/test_copy_resource.py

```diff
@@ -0,0 +1,294 @@
+                             msg="CellSizeYValue of new copied resource is not equal to "
+                                 "that of the original resource")
+            self.assertEqual(newcell.cellDataType, oldcell.cellDataType,
+                             msg="CellDataType of new copied resource is not equal to "
+                                 "that of the original resource")
+
+        if BandInformation.objects.filter(object_id=self.res_raster.metadata.id).exists():
+            self.assertTrue(BandInformation.objects.filter(
+                object_id=new_res_raster.metadata.id).exists())
+            newband = new_res_raster.metadata.bandInformation.first()
+            oldband = self.res_raster.metadata.bandInformation.first()
+            self.assertEqual(newband.name, oldband.name,
+                             msg="Band name of new copied resource is not equal to that of "
+                                 "the original resource")
+
+        # test to make sure a new unique identifier has been created for the new copied resource
+        self.assertIsNotNone(new_res_raster.short_id, msg='Unique identifier has not been '
+                                                          'created for new copied resource.')
+        self.assertNotEqual(new_res_raster.short_id, self.res_raster.short_id)
+
+        # test to make sure the new copied resource has 2 content file
+        # since an additional vrt file is created
+        self.assertEqual(new_res_raster.files.all().count(), 2)
+
+        # test to make sure the new copied resource has the correct identifier
+        self.assertEqual(new_res_raster.metadata.identifiers.all().count(), 1,
+                         msg="Number of identifier elements not equal to 1.")
+        self.assertIn('hydroShareIdentifier',
+                      [id.name for id in new_res_raster.metadata.identifiers.all()],
+                      msg="hydroShareIdentifier name was not found for new copied resource.")
+        id_url = '{}/resource/{}'.format(hydroshare.utils.current_site_url(),
+                                         new_res_raster.short_id)
+        self.assertIn(id_url, [id.url for id in new_res_raster.metadata.identifiers.all()],
+                      msg="Identifier url was not found for new copied resource.")
+
+        # test to make sure the new copied resource is linked with the original resource via
+        # isDerivedFrom Source metadata element
+        self.assertGreater(new_res_raster.metadata.sources.all().count(), 0,
+                           msg="New copied resource does not has source element.")
+
+        derived_from_value = '{}/resource/{}'.format(hydroshare.utils.current_site_url(),
+                                                     self.res_raster.short_id)
+        self.assertIn(derived_from_value,
+                      [src.derived_from for src in new_res_raster.metadata.sources.all()],
```

Comment:
I think since the resource would have only one ```source``` element, it probably be better to use ```assertEqual```